File: 204.CountPrimes/Count.java
public class Solution {
    public int countPrimes(int n) {
        if (n < 3) {
            return 0;
        }

        int[] prime = new int[n];
        prime[0] = prime[1] = 1;

        for (int i = 2; i < (int)(Math.sqrt(n))+1; i++) {
            if (prime[i] == 0) {
                for (int j = i*i; j < n; j += i) {
                    prime[j] = 1;
                }
            }
        }

        int sum = 0;
        for (int i = 0; i < n; i++) {
            if (prime[i] == 0) {
                sum++;
            }
        }

        return sum;
    }
}

File: 204.CountPrimes/count.py
from math import sqrt

class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n <= 2: return 0

        primes = [1] * n
        primes[0] = primes[1] = 0

        for i in range(2, int(n**0.5)+1):
            if primes[i]:
                primes[i*i:n:i] = [0] * len(primes[i*i:n:i])

        return sum(primes)


if __name__ == '__main__':
    n = 14
    print(Solution().countPrimes(n))
